applications/FluidDynamicsApplication/custom_processes/two_fluids_inlet_process.cpp

```diff
@@ -0,0 +1,243 @@
+    mrInletModelPart.CreateSubModelPart("fluid_2_inlet");
+    ModelPart& rWaterInlet = mrInletModelPart.GetSubModelPart("fluid_1_inlet");
+    ModelPart& rAirInlet = mrInletModelPart.GetSubModelPart("fluid_2_inlet");
+
+    // classifying nodes (no OMP parallel possible)
+    std::vector<IndexType> index_node_water;
+    std::vector<IndexType> index_node_air;
+    for (int i_node = 0; i_node < static_cast<int>( mrInletModelPart.NumberOfNodes() ); ++i_node){
+        auto it_node = mrInletModelPart.NodesBegin() + i_node;
+        const double inlet_dist = ComputeNodalDistanceInInletDistanceField(it_node);
+        if (inlet_dist <= 0.0){
+            index_node_water.push_back( it_node->GetId() );
+        } else {
+            index_node_air.push_back( it_node->GetId() );
+        }
+    }
+    rWaterInlet.AddNodes( index_node_water );
+    rAirInlet.AddNodes( index_node_air );
+
+    // classifying conditions (no OMP parallel possible)
+    std::vector<IndexType> index_cond_water;
+    std::vector<IndexType> index_cond_air;
+    for (int i_cond = 0; i_cond < static_cast<int>( mrInletModelPart.NumberOfConditions() ); ++i_cond){
+        auto it_cond = mrInletModelPart.ConditionsBegin() + i_cond;
+        unsigned int pos_counter = 0;
+        unsigned int neg_counter = 0;
+        for (int i_node = 0; i_node < static_cast<int>(it_cond->GetGeometry().PointsNumber()); i_node++){
+            const Node<3>& rNode = (it_cond->GetGeometry())[i_node];
+            const double inlet_dist = ComputeNodalDistanceInInletDistanceField( rNode );
+            if ( inlet_dist > 0 ){ pos_counter++; }
+            if ( inlet_dist <= 0 ){ neg_counter++; }
+        }
+        // the conditions cut by the interface are neither assigned to both the positive and the negative side
+        if( pos_counter > 0 ){
+            index_cond_air.push_back( it_cond->GetId() );
+        }
+        if( neg_counter > 0 ){
+            index_cond_water.push_back( it_cond->GetId() );
+        }
+    }
+    rWaterInlet.AddConditions( index_cond_water );
+    rAirInlet.AddConditions( index_cond_air );
+
+    rRootModelPart.GetCommunicator().Barrier();
+}
+
+
+
+void TwoFluidsInletProcess::SmoothDistanceField(){
+
+    ModelPart& rRootModelPart = mrInletModelPart.GetRootModelPart();
+
+    // #pragma omp parallel for
+    for (int i_node = 0; i_node < static_cast<int>( rRootModelPart.NumberOfNodes() ); ++i_node){
+        // iteration over all nodes
+        auto it_node = rRootModelPart.NodesBegin() + i_node;
+
+        // check if node is inside "inlet_transition_radius"
+        if ( std::abs(it_node->GetValue(AUX_DISTANCE)) > 1.0e-5 ){
+            // finding distance value of the node in the inlet field
+            const double inlet_dist = ComputeNodalDistanceInInletDistanceField(it_node);
+            // finding distance value for the node in the domain distance field
+            const double domain_dist = it_node->FastGetSolutionStepValue( DISTANCE, 0 );
```

Comment:
Use a const reference instead (memory consumption).

Reply:
Makes sense. Thank you. Done.